lib/ecto/adapters/sql.ex

```diff
@@ -922,7 +926,7 @@ defmodule Ecto.Adapters.SQL do
 
     all_params = placeholders ++ Enum.reverse(params, conflict_params)
 
-    %{num_rows: num, rows: rows} = query!(adapter_meta, sql, all_params, opts)
+    %{num_rows: num, rows: rows} = query!(adapter_meta, sql, all_params, put_source(opts, source))
```

Comment:
We don't need to call put_source because it is always present, right?

```suggestion
    %{num_rows: num, rows: rows} = query!(adapter_meta, sql, all_params, [source: source] ++ opts)
```